### PR TITLE
Bump fastcrypto

### PR DESCRIPTION
**Pull request by jonas-lj**
Sorry for the large PR, but there have been a few changes to the fastcrypto api, so in order to bump the version, we need to change a few things in sui. It's mostly just changes in paths and small changes to the api.

The fastcrypto version in mysten-util-mem (and afterwards, the mysten-util-mem depdency here) needs to be updated before this is merged.

Unintentionally, I've had to update the openrpc.json schema in order for the test_json_rpc_spec test to pass. I'm not sure why this is necessary, but it may be due to some updated dependencies pulled in from fastcrypto.

### CODE REVIEW

File: crates/sui-open-rpc/spec/openrpc.json
Comment (jonas-lj) [inline crates/sui-open-rpc/spec/openrpc.json:3088]: I don't know why it was necessary to update the schema here, but it could be because of some updated dependencies.

**Review by kchalkias — COMMENTED** [review]
Thanks @jonas-lj , we generally prefer rebasing vs merging to main. We will merge anyway after the PR is approved.

**Review by kchalkias — APPROVED** [review]

**Comment by jonas-lj** [comment]
> Thanks @jonas-lj , we generally prefer rebasing vs merging to main. We will merge anyway after the PR is approved.

Yea, Francois reminded me. I'll make sure to do that in the future. And thanks for the review!

**Pull request merged**
